### PR TITLE
Prefill with business name and type if they exist, change email domains

### PR DESCRIPTION
**Pull request by jojo-stripe**
Disputes now have a customer
image

Business name and type are prefilled if they are specified:

Individual (taken straight to identity info page):
https://github.com/user-attachments/assets/b32e74af-db84-4c13-b764-a118f349055f
Other (have to pick)
https://github.com/user-attachments/assets/7f845753-d4b0-446d-a8bd-a88896b6e6da

Also tested with company but did not screen record

### CODE REVIEW

File: lib/auth.ts [hunk 1]
```diff
@@ -381,10 +382,23 @@ export const authOptions: AuthOptions = {
             console.log('Could not find an existing user for the email', email);
             return null;
           }
+          let businessType = undefined;
+
+          if (
```
Comment (jorgea-stripe) [inline lib/auth.ts:387]: This works, but would be good to do with a `switch` statement, and to leverage a union type that has the possible values of `businessType`, to ensure we are handling all cases. Like so:

```
switch (credentials?.businessType) {
  case 'company':
  case 'individual':
   return credentials?.businessType;
  default:
    assertUnreachable(credentials?.businessType);
    return 'individual'; // We default to individual, but rely on TS to catch missing cases
}
```

**Review by jorgea-stripe — APPROVED** [review]
Changes LGTM, left one suggestion

File: app/api/setup_accounts/route.ts [hunk 2]
```diff
@@ -63,6 +63,7 @@ export async function POST() {
         },
         confirm: true,
         payment_method: 'pm_card_createDispute',
+        receipt_email: 'dispute_test@stripe.com',
```
[the remark this answers is not in the document]
Comment (jojo-stripe) [reply app/api/setup_accounts/route.ts:66]: The picture in the description is a reference to this change!

File: lib/auth.ts (hunk 1 above)
Comment (jojo-stripe) [reply lib/auth.ts]: Added!

File: lib/auth.ts [hunk 3]
```diff
@@ -4,11 +4,16 @@ import dbConnect from '@/lib/dbConnect';
 import Salon from '../app/models/salon';
 import {stripe} from '@/lib/stripe';
 import {resolveControllerParams} from './utils';
+import Stripe from 'stripe';
 
 function getRandomInt(min: number, max: number) {
   return Math.floor(Math.random() * (max - min + 1)) + min;
 }
 
+function assertUnreachable(option: string) {
```
Comment (jorgea-stripe) [inline lib/auth.ts:13]: This should be typed with `never`, so that the switch statement gives a compile time error when there is a new option

**Review by jorgea-stripe — APPROVED** [review]

**Pull request merged**